File: backend/app/routes/token.py
from fastapi import APIRouter, Body
from app.services import token

router = APIRouter(prefix="/token", tags=["token"])

token_service = token.TokenService()

@router.post("/create")
def create_token(name: str = Body(...)):
    token_id = token_service.create_token(name)
    return {"status": "ok", "token_id": token_id}

@router.post("/move/{token_id}/{new_x}_{new_y}")
def move_token(token_id: int, new_x: int, new_y: int):
    if token_service.move_token(token_id, new_x, new_y):
        return {"status": "ok"}
    return {"status": "error", "message": "Invalid token ID or position out of bounds"}

@router.delete("/delete/{token_id}")
def delete_token(token_id: int):
    if token_service.delete_token(token_id):
        return {"status": "ok"}
    return {"status": "error", "message": "Token not found"}

@router.post("/rename/{token_id}")
def rename_token(token_id: int, new_name: str = Body(...)):
    if token_service.rename_token(token_id, new_name):
        return {"status": "ok"}
    return {"status": "error", "message": "Token not found"}

File: frontend/src/api/token.ts
import { Token } from '../models/token';

const BASE_URL = "http://localhost:8000"

// Cria um token
export async function createToken(name: string): Promise<number> {
    const response = await fetch(`${BASE_URL}/token/create`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ name })
    });
    const data = await response.json();
    if (data.status !== 'ok') throw new Error(data.message);
    return data.token_id;
}

// Move token
export async function moveToken(tokenId: number, newX: number, newY: number) {
    const response = await fetch(`${BASE_URL}/token/move/${tokenId}/${newX}_${newY}`, {
        method: 'POST'
    });
    const data = await response.json();
    if (data.status !== 'ok') throw new Error(data.message);
}

// Delete token
export async function deleteToken(tokenId: number) {
    const response = await fetch(`${BASE_URL}/token/delete/${tokenId}`, {
        method: 'DELETE'
    });
    const data = await response.json();
    if (data.status !== 'ok') throw new Error(data.message);
}

// Rename token
export async function renameToken(tokenId: number, newName: string) {
    const response = await fetch(`${BASE_URL}/token/rename/${tokenId}`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ new_name: newName })
    });
    const data = await response.json();
    if (data.status !== 'ok') throw new Error(data.message);
}
